### PR TITLE
Support custom provenance and source from SQL based instance

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

**Review by keyurva — APPROVED** [review]
Very cool!

File: internal/server/v2/observation/helper.go [hunk 1]
```diff
@@ -40,65 +40,68 @@ func handleSQLRows(
...
+					}
+				}
+				result.ByVariable[variable].ByEntity[entity].OrderedFacets = append(
+					result.ByVariable[variable].ByEntity[entity].OrderedFacets,
+					&pbv2.FacetObservation{
+						FacetId:      provID,
+						Observations: obsList,
+						ObsCount:     int32(len(obsList)),
+						EarliestDate: obsList[0].Date,
+						LatestDate:   obsList[len(obsList)-1].Date,
```
[the remark this answers is not in the document]
Comment (shifucun) [reply internal/server/v2/observation/helper.go:100]: Yes, the query sort the result by date

File: test/setup.go [hunk 2]
```diff
@@ -154,6 +155,13 @@ func setupInternal(
 		}
 	} else {
 		c = &resource.Cache{}
+		if useSQLite {
+			customProvenance, err := query.GetProvenance(st.SQLClient)
+			if err != nil {
+				return nil, err
+			}
+			c.CustomProvenance = customProvenance
```
[the remark this answers is not in the document]
Comment (shifucun) [reply test/setup.go]: Done

File: internal/sqldb/query/get_provenance.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+package query
+
+import (
+	"database/sql"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+)
+
+// GetProvenance returns all the provenance name and url in SQL database.
+func GetProvenance(sqlClient *sql.DB) (map[string]*pb.Facet, error) {
```
[the remark this answers is not in the document]
Comment (shifucun) [reply internal/sqldb/query/get_provenance.go]: Done

**Review by shifucun — COMMENTED** [review]
thanks for review!

**Pull request merged**